packages/zowe-explorer/src/trees/dataset/DatasetFSProvider.ts

```diff
@@ -604,11 +588,15 @@ export class DatasetFSProvider extends BaseProvider implements vscode.FileSystem
         const newName = path.posix.basename(newUri.path);
 
         try {
-            if (FsDatasetsUtils.isPdsEntry(entry)) {
+            if (FsDatasetsUtils.isPdsEntry(entry) || !entry.isMember) {
```

Comment:
Interesting! 🥳 
Nice to see some smarts built-in into the language.

It makes sense then to access the `isMember` property directly to avoid type guard checking, thus avoid having to type-case `entry` in the else condition 🙏🏽 

Thanks for explaining! 🙏🏽